platforms/documentation/docs/src/docs/userguide/dep-man/dependency-management/resolution_rules.adoc

```diff
@@ -0,0 +1,474 @@
+----
+configurations.all {
+    resolutionStrategy.dependencySubstitution {
+        all {
+            if (requested is ModuleComponentSelector && requested.group == "org.example" && requested.version == "1.0") {
+                useTarget(module("org.example:library:1.0")).because("Switching from platform to library variant")
+            }
+        }
+    }
+}
+----
+
+- The substitution is based on the requested dependency's attributes (like group and version).
+- This approach allows you to switch from a platform component to a library or vice versa.
+- It uses Gradle's variant-aware engine to ensure the correct variant is selected based on the configuration and consumer attributes.
+
+This flexibility is often required when working with complex dependency graphs where different component types (platforms, libraries) need to be swapped dynamically.
+
+[[sec:substitution_with_attributes]]
+==== Substituting a dependency with attributes
+
+Substituting a dependency based on attributes allows you to override the default selection of a component by targeting specific attributes (like platform vs. regular library).
+
+This technique is useful for managing platform and library dependencies in complex builds, particularly when you want to consume a regular library but the platform dependency was incorrectly declared.
+
+Let's look at an inccorect:
+
+====
+include::sample[dir="snippets/dependencyManagement/customizingResolution-attributeSubstitutionRule/kotlin",files="lib/build.gradle.kts[tags=dependencies]"]
+include::sample[dir="snippets/dependencyManagement/customizingResolution-attributeSubstitutionRule/groovy",files="lib/build.gradle[tags=dependencies]"]
+====
+
+In this example, the substitution rule targets the platform version of `com.google.guava:guava` and replaces it with the regular library version:
+
+====
+include::sample[dir="snippets/dependencyManagement/customizingResolution-attributeSubstitutionRule/kotlin",files="consumer/build.gradle.kts[tags=substitution_rule]"]
+include::sample[dir="snippets/dependencyManagement/customizingResolution-attributeSubstitutionRule/groovy",files="consumer/build.gradle[tags=substitution_rule]"]
+====
+
+Without the `platform` keyword, the substitution would not specifically target the platform dependency.
+
+The following rule performs the same substitution but uses the more granular **variant** notation, allowing for customization of the dependency's attributes:
+
+====
+include::sample[dir="snippets/dependencyManagement/customizingResolution-attributeSubstitutionRule/kotlin",files="consumer/build.gradle.kts[tags=substitution_rule_alternative]"]
+include::sample[dir="snippets/dependencyManagement/customizingResolution-attributeSubstitutionRule/groovy",files="consumer/build.gradle[tags=substitution_rule_alternative]"]
+====
+
+By using attribute-based substitution, you can precisely control which dependencies are replaced, ensuring Gradle resolves the correct versions and variants in your build.
```

Comment:
I think this example shouldn't use the `Platform` attribute, as it seems like a contrived example and distracts from the core point.  I think showing the possiblity of substituting variants should just substitute a dependency on something with a `color=red` with `color=blue` attribute, that is totally arbitrary and abstract.

Reply:
For the sake of getting this in 8.11, I will update this example in the next revision.